manifests/keypair.pp

```diff
@@ -0,0 +1,41 @@
+define certs::keypair (
+  $key_pair,
+  $key_file,
+  $cert_file,
+  $manage_key  = false,
+  $key_owner   = undef,
+  $key_group   = undef,
+  $key_mode    = undef,
+  $manage_cert = false,
```

Comment:
I'm unsure because I didn't know the larger impact of it. I also considered not adding the parameter at all and manage the file if any property is not undef.

Maybe we should do it in stages and start with this refactoring and go from there. I was talking to @timogoebel and we both think we can pull many service integrations from this into other modules. Let's see it on a case-by-case basis.